src/pages/NavigateToActiveAssessment/index.tsx

```diff
@@ -0,0 +1,108 @@
+import { Navigate } from 'react-router-dom';
+
+import { ActivityPipelineType, getDataFromProgressId, getProgressId } from '~/abstract/lib';
+import { RequestHealthRecordDataItemStep } from '~/entities/activity';
+import { appletModel } from '~/entities/applet';
+import { actions } from '~/entities/applet/model';
+import { useActiveAssessment } from '~/entities/applet/model/hooks';
+import { selectGroupProgress } from '~/entities/applet/model/selectors';
+import ROUTES from '~/shared/constants/routes';
+import { useAppDispatch, useAppSelector } from '~/shared/utils';
+
+const NavigateToActiveAssessment = () => {
+  const dispatch = useAppDispatch();
+  const { activeAssessment } = useActiveAssessment();
+
+  const groupProgress = useAppSelector((state) => {
+    if (!activeAssessment?.groupProgressId) return null;
+    return selectGroupProgress(state, activeAssessment.groupProgressId);
+  });
+
+  const progressData = activeAssessment?.groupProgressId
+    ? getDataFromProgressId(activeAssessment.groupProgressId)
+    : null;
+
+  const activityId =
+    groupProgress &&
+    progressData &&
+    (groupProgress.type === ActivityPipelineType.Flow
+      ? groupProgress.currentActivityId
+      : progressData.entityId);
+
+  const activityProgress = useAppSelector((state) => {
+    if (!activityId) return null;
+
+    return appletModel.selectors.selectActivityProgress(
+      state,
+      getProgressId(activityId, progressData.eventId, progressData.targetSubjectId),
+    );
+  });
+
+  /* No active assessment found, navigate to applet list
+  =================================================== */
+  if (!activeAssessment || !activityId) {
+    return <Navigate to={ROUTES.appletList.path} />;
+  }
+
+  /* Specific handling of activities containing EHR item types:
+   *
+   * When the activity contains an EHR item type that's in the OneUpHealth step, and we're resuming
+   * the activity via the /active-assessment route, advance the item's state to the last sub-step
+   * (after the OneUpHealth step). Also reset the additional EHRs request to unanswered.
```

Comment:
Very helpful for describing this scenario that's unique to EHR item types. Thanks!